akka-docs/rst/scala/remoting-artery.rst

```diff
@@ -0,0 +1,465 @@
+
+  "com.typesafe.akka" %% "akka-remote" % "@version@" @crossString@
+
+To enable remote capabilities in your Akka project you should, at a minimum, add the following changes
+to your ``application.conf`` file::
+
+  akka {
+    actor {
+      provider = remote
+    }
+    remote {
+      artery {
+        enabled = on
+        canonical.hostname = "127.0.0.1"
+        canonical.port = 2552
+      }
+    }
+  }
+
+As you can see in the example above there are four things you need to add to get started:
+
+* Change provider from ``local`` to ``remote``
+* Enable Artery to use it as the remoting implementation
+* Add host name - the machine you want to run the actor system on; this host
+  name is exactly what is passed to remote systems in order to identify this
+  system and consequently used for connecting back to this system if need be,
+  hence set it to a reachable IP address or resolvable name in case you want to
+  communicate across the network.
+* Add port number - the port the actor system should listen on, set to 0 to have it chosen automatically
+
+.. note::
+
+  The port number needs to be unique for each actor system on the same machine even if the actor
+  systems have different names. This is because each actor system has its own networking subsystem
+  listening for connections and handling messages as not to interfere with other actor systems.
+
+The example above only illustrates the bare minimum of properties you have to add to enable remoting.
+All settings are described in :ref:`remote-configuration-scala`.
+
+Canonical address
+^^^^^^^^^^^^^^^^^
+
+In order to remoting to work properly, where each system can send messages to any other system on the same network
+(for example a system forwards a message to a third system, and the third replies directly to the sender system)
+it is essential for every system to have a *unique, globally reachable* address and port. This address is part of the
+unique name of the system and will be used by other systems to open a connection to it and send messages. This means
+that if a host has multiple names (different DNS records pointing to the same IP address) then only one of these
+can be *canonical*. If a message arrives to a system but it contains a different hostname than the expected canonical
+name then the message will be dropped. If multiple names for a system would be allowed, then equality checks among
+:class:`ActorRef` instances would no longer to be trusted and this would violate the fundamental assumption that
+an actor has a globally unique reference on a given network.
+
+In cases, where Network Address Translation (NAT) is used or other network bridging is involved, it is important
+to configure the system so that it understands that there is a different between his externally visible, canonical
+address and between the host-port pair that is used to listen for connections. See :ref:`remote-configuration-nat-artery`
+for details.
+
```

Comment:
perhaps mention that 127.0.01 or localhost cannot be used if more than that machine is involved, i.e. fine for development but not production